src/layer.rs

```diff
@@ -2088,7 +2094,98 @@ mod tests {
         assert_eq!(child1.parent_span_id, root.span_context.span_id());
         assert_eq!(child2.parent_span_id, child1.span_context.span_id());
 
-        // This is surprising, the parent should be `child1`, but is 'root'.
+        // The parent should be `child1`
         assert_eq!(child3.parent_span_id, child1.span_context.span_id());
     }
+
+    #[test]
+    fn follows_from_adds_link() {
+        use crate::OpenTelemetrySpanExt;
+        let mut tracer = TestTracer::default();
+        let subscriber = tracing_subscriber::registry().with(layer().with_tracer(tracer.clone()));
+
+        let span1_id = tracing::subscriber::with_default(subscriber, || {
+            let span2 = tracing::debug_span!("span2");
+            let span1 = tracing::debug_span!("span1");
+
+            // Ensure that span2 is started
+            let _ = span2.context();
+
+            // Establish follows_from relationship
+            span2.follows_from(&span1);
+
+            // Enter span2 to ensure it's exported
+            let _guard = span2.enter();
+
+            // Get span ID for later verification
+            span1.context().span().span_context().span_id()
+        });
+
+        // Verify that span2 has a link to span1
+        let links = tracer.with_data_for("span2", |data| {
+            data.links
+                .as_ref()
+                .into_iter()
+                .map(|link| link.span_context.span_id())
+                .collect::<Vec<_>>()
+        });
+
+        assert_eq!(
+            links.len(),
+            1,
+            "Expected span to have one link from follows_from relationship"
+        );
+
+        assert!(
+            links.contains(&span1_id),
+            "Link should point to the correct source span"
+        );
+    }
```

Comment:
Because tracing-otel is _a bit_ complicated, should we add an assertion to check that both span1 and span2 are exported ? Said otherwise, that adding this link hasn't subtly broken something else with either involved span.

Reply:
Are there any other "externally viewed" requirements for follows- from to work?
I would assume there's no export order requirement - distributed systems and all that. Anything else?

Reply:
Sure, I can check that we have both spans. And what do you know, the spec is [a bit too prescriptive](https://opentelemetry.io/docs/specs/otel/trace/api/#link).

> Span SHOULD preserve the order in which Links are set.